baggage/baggage.go

```diff
@@ -304,9 +304,32 @@ func parseMember(member string) (Member, error) {
 	if err != nil {
 		return newInvalidMember(), fmt.Errorf("%w: %v", errInvalidValue, err)
 	}
+
+	if !utf8.ValidString(value) {
+		// W3C baggage spec:
+		// https://github.com/w3c/baggage/blob/main/baggage/HTTP_HEADER_FORMAT.md?plain=1#L69
+		for _, is := range findInvalidUTF8Sequences(value) {
+			value = strings.ReplaceAll(value, is, "�")
+		}
+	}
+
 	return Member{key: key, value: value, properties: props, hasData: true}, nil
 }
 
+func findInvalidUTF8Sequences(input string) []string {
+	var invalidSequences []string
+	for i := 0; i < len(input); i++ {
+		r, size := utf8.DecodeRuneInString(input[i:])
+
+		if r == utf8.RuneError && size == 1 {
+			// RuneError indicates an invalid UTF-8 sequence
+			invalidSequences = append(invalidSequences, input[i:i+size])
```

Comment:
Should the for loop skip to `i+size` rather than `i++`?

Reply:
You are right. Fixed it.